### PR TITLE
Update output recording for compatibility

### PR DESCRIPTION
**Pull request by swernli**
This change returns the legacy output recording functions to allow for broader compatibility across different bitcode patterns. If the newer output recording functions are mixed with the legacy ones, an error is raised, so programs must either be all legacy style or all new style.

Additionally, function binding now checks parameter length across the board to ensure the right bindings occur and reduce the likelihood of access violations in JIT compiled QIR.

### CODE REVIEW

File: runner/src/lib.rs [hunk 1]
```diff
@@ -72,7 +72,9 @@ fn run_module(module: &Module, entry_point: Option<&str>) -> Result<(), String>
         .create_jit_execution_engine(OptimizationLevel::None)
         .map_err(|e| e.to_string())?;
 
-    bind_functions(module, &execution_engine)?;
+    if !bind_functions(module, &execution_engine)? {
+        println!("WARNING: Mix of legacy and current output recording functions found. Please update your tooling.");
+    }
```
Comment (idavis) [inline runner/src/lib.rs:77]: We should undo this change along with the bool return of bind_functions with bool. The code that returns bool should return Err() with the message here.

**Review by idavis — CHANGES_REQUESTED** [review]

File: runner/src/lib.rs (hunk 1 above)
Comment (swernli) [reply runner/src/lib.rs:77]: Updated!

**Review by idavis — APPROVED** [review]
